src/Form/BaseSettingsForm.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+namespace Drupal\fb_instant_articles\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Facebook Instant Articles base settings form.
+ */
+class BaseSettingsForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'fb_instant_articles.base_settings',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'base_settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('fb_instant_articles.base_settings');
+
+    // Add the page id configuration.
+    $args = [
+      '@claim_url' => 'https://developers.facebook.com/docs/instant-articles/claim-url',
+    ];
+    $form['page_id'] = [
+      '#title' => t('Facebook Page ID'),
+      '#type' => 'textfield',
+      '#default_value' => $config->get('page_id'),
+      '#required' => TRUE,
+      '#description' => $this->t('In order to designate the domain that will host your
```

Comment:
I'm pretty sure this has to all be on one line for localize.drupal.org to be able to extract the string.